testplan/common/utils/process.py

```diff
@@ -229,7 +231,7 @@ def execute_cmd(
     start_time = time.time()
 
     handler = subprocess.Popen(cmd, stdout=stdout, stderr=stderr, env=env)
-    stdout, stderr = handler.communicate()
+    stdout, stderr = handler.communicate(timeout=timeout)
```

Comment:
Should not the subprocess being cleaned killed, and logs collected in case of timeout? This might be able to leave behind running processes. The outer scope would not have the handle to clean up.

Reply:
i agree, i have removed this change ;)